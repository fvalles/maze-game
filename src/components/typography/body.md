Basic body text component with default color="text":

```jsx
<Body>Lorem ipsum</Body>
```

Body text component with color="white":

```jsx
<Body color="white">Lorem ipsum</Body>
```
