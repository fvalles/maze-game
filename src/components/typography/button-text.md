Basic button text component with default color="white":

```jsx
<ButtonSmall>Lorem ipsum</ButtonSmall>
```

Button text component with color="text":

```jsx
<ButtonSmall color="text">Lorem ipsum</ButtonSmall>
```
