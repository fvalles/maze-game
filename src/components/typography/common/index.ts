import { KeyColors } from '../../../core/theme/colors';

/**
 * Types
 */

export interface TypographyProps {
  /** Text color */
  color?: KeyColors;
}
