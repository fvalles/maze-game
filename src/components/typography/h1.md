Basic h1 text component with default color="text":

```jsx
<H1>Lorem ipsum</H1>
```

H1 text component with color="white":

```jsx
<H1 color="white">Lorem ipsum</H1>
```
