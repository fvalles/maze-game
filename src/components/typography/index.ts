export { Body } from './body';
export { ButtonText } from './button-text';
export { H1 } from './h1';
export { H2 } from './h2';
