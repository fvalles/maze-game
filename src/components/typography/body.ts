import styled from 'styled-components';
import { TypographyProps } from './common';

export const Body = styled.p<TypographyProps>`
  color: ${({ color = 'text', theme }) => theme.Colors[color]};
  font-family: ${({ theme }) => theme.Fonts.light}, 'Avenir Next', 'Segoe UI',
    sans-serif;
  font-size: 20px;
  line-height: 24px;
`;
