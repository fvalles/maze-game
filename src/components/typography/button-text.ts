import styled from 'styled-components';
import { TypographyProps } from './common';

export const ButtonText = styled.span<TypographyProps>`
  color: ${({ color = 'white', theme }) => theme.Colors[color]};
  font-family: ${({ theme }) => theme.Fonts.regular}, 'Avenir Next', 'Segoe UI',
    sans-serif;
  font-size: 20px;
  font-weight: bold;
  line-height: 16px;
`;
