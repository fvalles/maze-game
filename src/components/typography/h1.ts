import styled from 'styled-components';
import { TypographyProps } from './common';

export const H1 = styled.h1<TypographyProps>`
  color: ${({ color = 'text', theme }) => theme.Colors[color]};
  font-family: ${({ theme }) => theme.Fonts.regular}, 'Avenir Next', 'Segoe UI',
    sans-serif;
  font-size: 58px;
  line-height: 62px;
`;
