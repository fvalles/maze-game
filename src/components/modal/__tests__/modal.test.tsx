import React from 'react';
import { ThemeProvider } from '@core/theme/theme-provider';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { H1 } from '@components/typography';
import { Modal } from '..';

jest.mock('react-dom', () => ({
  ...jest.requireActual('react-dom'),
  createPortal: (node: any) => node,
}));

const MODAL_DIMENSION = 400;

describe('modal component', () => {
  it('should not render if isOpen property is false', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <Modal
            height={MODAL_DIMENSION}
            isOpen={false}
            label="some label"
            testId="some-test-id"
            width={MODAL_DIMENSION}>
            <H1>Lorem ipsum</H1>
          </Modal>
        </ThemeProvider>,
      )
      .toJSON();

    expect(result).toBe(null);
    expect(result).toMatchSnapshot();
  });

  it('should render if isOpen property is true', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <section>
            <Modal
              height={MODAL_DIMENSION}
              isOpen
              label="some label"
              testId="some-test-id"
              width={MODAL_DIMENSION}>
              <H1>Lorem ipsum</H1>
            </Modal>
          </section>
        </ThemeProvider>,
      )
      .toJSON();

    const container = result.children[0];
    const modal = container.children[0];
    const modalLabel = modal.props['aria-label'];
    expect(modalLabel).toBe('some label');

    const modalTextContent = modal.children[0];
    expect(modalTextContent.type).toBe('h1');
    expect(modalTextContent.children[0]).toBe('Lorem ipsum');
    expect(result).toMatchSnapshot();
  });
});
