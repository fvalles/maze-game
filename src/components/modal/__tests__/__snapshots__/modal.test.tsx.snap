// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal component should not render if isOpen property is false 1`] = `null`;

exports[`modal component should render if isOpen property is true 1`] = `
.c0 {
  color: #212240;
  font-family: Avenir Regular,'Avenir Next','Segoe UI', sans-serif;
  font-size: 58px;
  line-height: 62px;
}

<section>
  <div
    className="ReactModal__Overlay"
    onClick={[Function]}
    onMouseDown={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.75)",
        "bottom": 0,
        "left": 0,
        "position": "fixed",
        "right": 0,
        "top": 0,
      }
    }
  >
    <div
      aria-label="some label"
      aria-modal={true}
      className="ReactModal__Content"
      data-testid="some-test-id"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      role="dialog"
      style={
        Object {
          "WebkitOverflowScrolling": "touch",
          "alignItems": "center",
          "background": "#fff",
          "backgroundColor": "#F9F2ED",
          "border": "1px solid #ccc",
          "borderRadius": 4,
          "bottom": "40px",
          "boxShadow": "2px 2px 5px, -2px 2px 5px, 2px -2px 5px, -2px -2px 5px",
          "display": "flex",
          "height": "400px",
          "justifyContent": "center",
          "left": "50%",
          "outline": "none",
          "overflow": "auto",
          "padding": "20px",
          "position": "absolute",
          "right": "40px",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
          "width": "400px",
        }
      }
      tabIndex="-1"
    >
      <h1
        className="c0"
      >
        Lorem ipsum
      </h1>
    </div>
  </div>
</section>
`;
